### PR TITLE
Adding fixes to pass hardened profile of CIS scan 

### PR DESCRIPTION
**Pull request by prachidamle**
1) Adding a post-install hook to patch the default serviceaccount of fleet-system namespace
2) Adding a default network_policy to the namespace to pass CIS

https://github.com/rancher/cis-operator/issues/11

### CODE REVIEW

File: charts/fleet-agent/templates/patch_default_serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "helm.sh/hook": post-install, post-upgrade
+    "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation
+spec:
+  template:
+    spec:
+      serviceAccountName: fleet-agent
+      restartPolicy: Never
+      containers:
+      - name: sa
+        image: "{{ template "system_default_registry" . }}{{ .Values.global.kubectl.repository }}:{{ .Values.global.kubectl.tag }}"
```
Comment (StrongMonkey) [inline charts/fleet-agent/templates/patch_default_serviceaccount.yaml:16]: Why patching it instead of adding it directly to the service account?

Comment (StrongMonkey) [reply charts/fleet-agent/templates/patch_default_serviceaccount.yaml:16]: oh nvm

Comment (prachidamle) [reply charts/fleet-agent/templates/patch_default_serviceaccount.yaml:16]: this is the default serviceaccount auto-created by kubernetes after helm creates the namespace

**Comment by maggieliu** [comment]
related also to: https://github.com/rancher/fleet/issues/94

**Review by ibuildthecloud — APPROVED** [review]

**Pull request merged**